src/nvim/eval.c

```diff
@@ -7102,14 +7102,27 @@ static int get_float_arg(typval_T *argvars, float_T *f)
   return FAIL;
 }
 
+// Apply a floating point C function on a typval with one float_T.
+static inline void float_op_wrapper(typval_T *argvars, typval_T* rettv,
+                                    float_T (*function)(float_T))
+{
+  float_T f;
+
+  rettv->v_type = VAR_FLOAT;
+  if (get_float_arg(argvars, &f) == OK) {
+    rettv->vval.v_float = function(f);
```

Comment:
Good catch! I've made `get_float_arg` static inline, and it does indeed seem to fix gcc's complaint. Thanks!